crates/language_model/src/request.rs

```diff
@@ -296,18 +297,50 @@ impl LanguageModelRequest {
 
             match message.role {
                 Role::User | Role::Assistant => {
+                    let cache_control = if message.cache {
+                        Some(anthropic::CacheControl {
+                            cache_type: anthropic::CacheControlType::Ephemeral,
+                        })
+                    } else {
+                        None
+                    };
+                    let anthropic_message_content: Vec<anthropic::Content> = message
+                        .content
+                        .into_iter()
+                        // TODO: filter out the empty messages in the message construction step
```

Comment:
This conflicts now but I started this in https://github.com/zed-industries/zed/pull/16270. Happy to finish it up.

Reply:
@rgbkrk that'd be great if you don't mind - once this lands I was also gonna take a run at enabling caching for the inline assistant prompt as well.

Reply:
Cool, I'll merge this now and then rebase my branch against yours.